docs/docsite/rst/user_guide/playbooks_filters.rst

```diff
@@ -1,121 +1,46 @@
-To avoid such behaviour and generate long lines it is possible to use ``width`` option::
-
-    {{ some_variable | to_yaml(indent=8, width=1337) }}
-    {{ some_variable | to_nice_yaml(indent=8, width=1337) }}
-
-While it would be nicer to use a construction like ``float("inf")`` instead of a hardcoded number, unfortunately the filter doesn't support proxying Python functions.
-Note that it also supports passing through other YAML parameters. Full list can be found in `PyYAML documentation`_.
-
-
-Alternatively, you may be reading in some already formatted data::
-
-    {{ some_variable | from_json }}
-    {{ some_variable | from_yaml }}
-
-for example::
-
-  tasks:
-    - shell: cat /some/path/to/file.json
-      register: result
-
-    - set_fact:
-        myvar: "{{ result.stdout | from_json }}"
-
-
-.. versionadded:: 2.7
-
-To parse multi-document yaml strings, the ``from_yaml_all`` filter is provided.
-The ``from_yaml_all`` filter will return a generator of parsed yaml documents.
-
-for example::
-
-  tasks:
-    - shell: cat /some/path/to/multidoc-file.yaml
-      register: result
-    - debug:
-        msg: '{{ item }}'
-      loop: '{{ result.stdout | from_yaml_all | list }}'
-
-
-.. _forcing_variables_to_be_defined:
-
-Forcing Variables To Be Defined
-```````````````````````````````
-
-The default behavior from ansible and ansible.cfg is to fail if variables are undefined, but you can turn this off.
-
-This allows an explicit check with this feature off::
-
-    {{ variable | mandatory }}
-
-The variable value will be used as is, but the template evaluation will raise an error if it is undefined.
+Handling undefined variables
+============================
 
+Filters can help you manage missing or undefined variables by providing defaults or allowing null values. If you configure Ansible to ignore most undefined variables, you can mark some variables as requiring values with the ``mandatory`` filter.
 
 .. _defaulting_undefined_variables:
 
-Defaulting Undefined Variables
-``````````````````````````````
+Providing default values
+------------------------
 
-Jinja2 provides a useful 'default' filter that is often a better approach to failing if a variable is not defined::
+You can provide default values for variables directly in your templates using the Jinja2 'default' filter. This is often a better approach than failing if a variable is not defined::
 
     {{ some_variable | default(5) }}
 
-In the above example, if the variable 'some_variable' is not defined, the value used will be 5, rather than an error
-being raised.
-
+In the above example, if the variable 'some_variable' is not defined, Ansibie uses the default value 5, rather than raising an "undefined variable" error and failing. If you are working within a role, you can also add a ``defaults/main.yml`` to define the default values for variables in your role.
```

Comment:
```suggestion
In the above example, if the variable 'some_variable' is not defined, Ansible uses the default value 5, rather than raising an "undefined variable" error and failing. If you are working within a role, you can also add a ``defaults/main.yml`` to define the default values for variables in your role.
```